libraries/support/azure/management.rb

```diff
@@ -102,9 +102,10 @@ def storage_account(resource_group, name)
       )
     end
 
-    def storage_accounts
+    def storage_accounts(resource_group)
       get(
-        url: link(location: 'Microsoft.Storage/storageAccounts'),
+        url: link(location: 'Microsoft.Storage/storageAccounts',
+                  resource_group: resource_group),
```

Comment:
So this is OK even when `resource_group` is `nil`? Are you sure it shouldn't be conditionally added based on whether it has a value?

Reply:
We check for `nil` here: https://github.com/inspec/inspec-azure/blob/master/libraries/support/azure/management.rb#L286

I also have a test that does the query with and without a `resource_group`. https://github.com/inspec/inspec-azure/pull/143/files#diff-5496d1f35534e24f5840421d4cad3a57